conans/test/unittests/client/tools/cppstd_required_test.py

```diff
@@ -0,0 +1,183 @@
+import unittest
+from mock import mock
+from parameterized import parameterized
+
+from conans.test.utils.conanfile import MockConanfile, MockSettings
+from conans.client.tools import OSInfo
+from conans.errors import ConanInvalidConfiguration
+
+from conans.tools import check_min_cppstd, valid_min_cppstd
+
+
+class UserProneTests(unittest.TestCase):
+
+    def setUp(self):
+        self.conanfile = MockConanfile(MockSettings({}))
```

Comment:
I would move this to each test function, it is just a line.

Reply:
usually I hear, duplication is evil, but okay, I gonna replicate it.

Reply:
nevermind, that member has been used only once in this class, you are right.

Reply:
done